### PR TITLE
[fix]: [CDS-42637]: imagePath blank check

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37972)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37972)

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gcr/GcrArtifactResource.java [hunk 1]
```diff
@@ -108,11 +108,15 @@ public ResponseDTO<GcrResponseDTO> getBuildDetails(@QueryParam("imagePath") Stri
         gcrConnectorIdentifier = gcrArtifactConfig.getConnectorRef().getValue();
       }
     }
+
     // todo(hinger): resolve other expressions here?
     IdentifierRef connectorRef =
         IdentifierRefHelper.getIdentifierRef(gcrConnectorIdentifier, accountId, orgIdentifier, projectIdentifier);
     imagePath = artifactResourceUtils.getResolvedImagePath(accountId, orgIdentifier, projectIdentifier,
         pipelineIdentifier, runtimeInputYaml, imagePath, fqnPath, gitEntityBasicInfo, serviceRef);
+    if (EmptyPredicate.isEmpty(imagePath)) {
```
Comment (puthrayaharness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/gcr/GcrArtifactResource.java:117]: Please move logic to the service layer. Also please add test case

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37972)* status: :shipit: complete! ✓

**Pull request merged**